### DISCUSSION TITLE
Camera location and rotation

### DISCUSSION THREAD

**Discussion by kevytmaito11** [Sixze/ALS-Refactored#209] [General]
Is there a reason why CameraLocation and CameraRotation are made as a protected values instead of public? I need to get the location and rotation from the blueprint to get aiming position and forward vector and only way I could achieve this was to change variables to public:

public:
	UPROPERTY(VisibleAnywhere, BlueprintReadOnly, Category = "State", Transient)
		FVector CameraLocation;

	UPROPERTY(VisibleAnywhere, BlueprintReadOnly, Category = "State", Transient)
		FRotator CameraRotation;

Is there an another way to get location and rotation than changing these variables to public?

**Comment by ameaninglessname** [comment]
`UAlsCameraComponent::GetViewInfo`

**Reply by kevytmaito11** [reply]
> `UAlsCameraComponent::GetViewInfo`

Thank you!

**Reply by miketoupin** [reply]
Please can you confirm where you put the "UAlsCameraComponent::GetViewInfo" in the C++ code? I have the same issues and I am not an expert in C++ but i still have some basic knowledge of it., Thx!

**Reply by catarrestor** [reply]
I also need help with this.

**Comment by Sixze** [comment]
You can use `UAlsCameraComponent::GetViewInfo()`, but it is more like an "internal" function that is only used to pass camera properties to Unreal's internal logic.

For your case, I think it would be better to add const getters for the variables you need, instead of making them public.

**Comment by Sixze** [comment]
You can also use `APlayerCameraManager` to access some other camera properties.

**Comment by kevytmaito11** [comment]
I used blueprint to get access to rotation and location like this:

![image](https://user-images.githubusercontent.com/10709995/222983953-070e9a0c-3db5-46d9-ac89-7899e2c8be6c.png)

**Reply by Sixze** [reply]
From blueprints, you can access `CameraRotation` and other similar variables directly, because they are `BlueprintReadOnly`, there is no need to call the `GetViewInfo` function.
